kernel/kernel-api/src/main/java/io/delta/kernel/internal/skipping/DataSkippingUtils.java

```diff
@@ -0,0 +1,270 @@
+                Expression right = getRight(dataFilters);
+
+                if (left instanceof Column && right instanceof Literal) {
+                    Column leftCol = (Column) left;
+                    Literal rightLit = (Literal) right;
+                    if (schemaHelper.isSkippingEligibleMinMaxColumn(leftCol) &&
+                        schemaHelper.isSkippingEligibleLiteral(rightLit)) {
+                        return Optional.of(constructComparatorDataSkippingFilters(
+                            dataFilters.getName(), leftCol, rightLit, schemaHelper));
+                    }
+                } else if (right instanceof Column && left instanceof Literal) {
+                    return constructDataSkippingFilter(reverseComparatorFilter(dataFilters), schemaHelper);
+                }
+            // TODO more expressions
+            default:
+                return Optional.empty();
+        }
+    }
+
+    //////////////////////////////////////////////////////////////////////////////////
+    // Useful helper functions for dealing with expressions
+    //////////////////////////////////////////////////////////////////////////////////
+
+    /** Construct the skipping predicate for a given comparator */
+    private static Predicate constructComparatorDataSkippingFilters(
+            String comparator, Column leftCol, Literal rightLit, StatsSchemaHelper schemaHelper) {
+
+        switch (comparator.toUpperCase(Locale.ROOT)) {
+
+            // Match any file whose min/max range contains the requested point.
+            case "=":
+                // For example a = 1 --> minValue.a <= 1 AND 1 <= maxValue.a
+                return new And(
+                    new Predicate("<=", schemaHelper.getMinColumn(leftCol), rightLit),
+                    new Predicate("<=", rightLit, schemaHelper.getMaxColumn(leftCol))
+                );
+
+            // Match any file whose min is less than the requested upper bound.
+            case "<":
+                return new Predicate("<", schemaHelper.getMinColumn(leftCol), rightLit);
+
+            // Match any file whose min is less than or equal to the requested upper bound
+            case "<=":
+                return new Predicate("<=", schemaHelper.getMinColumn(leftCol), rightLit);
+
+            // Match any file whose max is larger than the requested lower bound.
+            case ">":
+                return new Predicate(">", schemaHelper.getMaxColumn(leftCol), rightLit);
+
+            // Match any file whose max is larger than or equal to the requested lower bound.
+            case ">=":
+                return new Predicate(">=", schemaHelper.getMaxColumn(leftCol), rightLit);
+
+            default:
+                throw new IllegalArgumentException(
+                    String.format("Unsupported comparator expression %s", comparator));
+        }
+    }
+
+    private static Map<String, String> reverseComparators = new HashMap<String, String>(){
```

Comment:
`final`, too, since we don't want this to be re-assigned

Reply:
and then, of course, `REVERSE_COMPARATORS` to stick to java8 convention for final variables